dbms/include/DB/Functions/FunctionsMiscellaneous.h

```diff
@@ -1245,7 +1245,7 @@ class FunctionRunningAccumulate : public IFunction
 		const auto & states = column_with_states->getData();
 		for (const auto & state_to_add : states)
 		{
-			agg_func.merge(place.get(), state_to_add);
+			agg_func.merge(place.get(), state_to_add, nullptr); /// Empty arena!
```

Comment:
Or add 'bool allocatesMemoryInArena()' method to IAggregateFunction and throw exception for that functions here.